File: taskmanager/src/main/kotlin/io/miragon/domain/UserTask.kt
package io.miragon.domain

import java.time.Instant

data class UserTask(
    val key: Long,
    val elementId: String,
    val processInstanceKey: Long,
    val bpmnProcessId: String,
    val processDefinitionKey: Long,
    val variables: Map<String, Any> = emptyMap(),
    var expiresAt: Instant,
    var assignee: String?,
    var taskState: String = "CREATED"
)
{
    fun extendLock(until: Instant)
    {
        this.expiresAt = until
    }
}

File: apps/tasklist/src/model/UserTask.ts
interface UserTaskParameters {
    key: number;
    elementId: string;
    processInstanceKey: number;
    bpmnProcessId: string;
    processDefinitionKey: number;
    variables: { [key: string]: object };
    taskState: string;
    assignee?: string;
}

export class UserTask {

    readonly key: number;

    readonly elementId: string;

    readonly processInstanceKey: number;

    readonly bpmnProcessId: string;

    readonly processDefinitionKey: number;

    readonly variables: { [key: string]: object };

    readonly taskState: string;

    readonly assignee?: string;

    constructor({
                    key,
                    elementId,
                    processInstanceKey,
                    bpmnProcessId,
                    processDefinitionKey,
                    variables,
                    taskState,
                    assignee,
                }: UserTaskParameters) {
        this.key = key;
        this.elementId = elementId;
        this.processInstanceKey = processInstanceKey;
        this.bpmnProcessId = bpmnProcessId;
        this.processDefinitionKey = processDefinitionKey;
        this.variables = variables;
        this.taskState = taskState;
        this.assignee = assignee;
    }
}